opentracing-cdi-example/src/main/java/io/opentracing/contrib/cdi/example/SpanContextProducer.java

```diff
@@ -0,0 +1,18 @@
+package io.opentracing.contrib.cdi.example;
+
+import io.opentracing.SpanContext;
+import io.opentracing.contrib.web.servlet.filter.TracingFilter;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+public class SpanContextProducer {
+    @Inject
+    HttpServletRequest request;
+
+    @Produces
+    public SpanContext produceSpanContext() {
+        return (SpanContext) request.getAttribute(TracingFilter.SERVER_SPAN_CONTEXT);
```

Comment:
> (this would require whoever started the span to deactivate it)

..whoever ~~started~~ activated the span to deactivate it..
I think it is most reasonable that activation / deactivation are paired. i.e. there _could_ be an active span (or may not be). However, the one placing it there is also responsible for removing it.
(usually in a try .. with resources construct from some executorservice that propagates activespans).